### PR TITLE
feat: add support for changing the receiver’s viewport to fit given bounds

### PR DESCRIPTION
**Pull request by TimAlber**
Add a feature to fit the viewport to given coordinate bounds.

### CODE REVIEW

**Comment by m0nac0** [comment]
@TimAlber It would be great if you could also add a simple example to the example app. That makes it easier to test this and it also showcases the feature.

File: example/lib/given_bounds.dart [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+          onPressed: () async {
+            await mapController.setCameraBounds(
+              west: 5.98865807458,
+              south: 47.3024876979,
+              east: 15.0169958839,
+              north: 54.983104153,
+              padding: 25,
+            );
+          },
+          child: const Text('Set Bounds to Germanny'),
```
Comment (m0nac0) [inline example/lib/given_bounds.dart]:
```suggestion
          child: const Text('Set bounds to Germany'),
```

Comment (m0nac0) [reply example/lib/given_bounds.dart]: Just one nit ;)

**Comment by m0nac0** [comment]
Thank you for the example. Do you also want to add a web implementation? If not, I'll create an issue and someone can add it later on.

**Comment by TimAlber** [comment]
I think it makes most sense if you create an issue and someone can add it later on.

File: example/lib/given_bounds.dart (hunk 1 above)
Comment (m0nac0) [reply example/lib/given_bounds.dart]: Ah, I also left this tiny review in "pending" mode 🤦‍♂️

File: example/lib/given_bounds.dart [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+          onPressed: () async {
+            await mapController.setCameraBounds(
+              west: -18,
+              south: -40,
+              east: 54,
+              north: 40,
+              padding: 25,
+            );
+          },
+          child: const Text('Set Bounds to Afrika'),
```
Comment (m0nac0) [inline example/lib/given_bounds.dart]:
```suggestion
          child: const Text('Set bounds to Africa'),
```

File: example/lib/given_bounds.dart [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:maplibre_gl/mapbox_gl.dart';
+
+import 'page.dart';
+
+class GivenBoundsPage extends ExamplePage {
+  GivenBoundsPage()
+      : super(const Icon(Icons.map_sharp), 'Changing given Bounds');
```
Comment (m0nac0) [inline example/lib/given_bounds.dart]:
```suggestion
      : super(const Icon(Icons.map_sharp), 'Changing given bounds');
```

**Review by m0nac0 — COMMENTED** [review]
Ready to merge this (apart from the very small suggestions), once this PR is up to date with the main branch ;)

File: example/lib/given_bounds.dart (hunk 3 above)
Comment (TimAlber) [reply example/lib/given_bounds.dart:12]: Made those changes.

**Comment by TimAlber** [comment]
I fixed those typos.

**Comment by TimAlber** [comment]
I don't know why iOS doesn't build. Maybe try to restart it?

**Review by m0nac0 — APPROVED** [review]
Thank you again for contributing this feature!

**Comment by m0nac0** [comment]
@TimAlber Your contributions would also be interesting for the [flutter-mapbox-gl repo](https://github.com/flutter-mapbox-gl/maps) where this plugin was forked from. Would you like to "upstream" your changes? If not, someone else could also do it.

**Comment by TimAlber** [comment]
@m0nac0 I think it would be best if someone else could do that.

**Pull request merged**